src/heap.rs

```diff
@@ -0,0 +1,247 @@
+/*
+General implementation of a heap
```

Comment:
https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html
If the interface matches this, that would be ideal

Reply:
https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.from

Can you check if this works natively in Solana? I don't see too much of a reason for this data structure in particular to be difficult to read in from an arbitrary buffer